go-sdk/sdk/connection.go

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package sdk
+
+import (
+	"encoding/json"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/apache/airflow/go-sdk/pkg/api"
+)
+
+type Connection struct {
+	ID string
+
+	// The Connection type, as defined in Airflow
+	Type string
+
+	Host string
+	Port int
+
+	// Login/username of the connection. Optional, can be nil (and nil/not set is distinct from an empty login)
+	Login *string
+	// Password of the connection. Optional, can be nil.
+	Password *string
+
+	// The path. Called `Schema` in Airflow python code. For database connections this often contains the
+	// database name.
+	Path string
```

Comment:
Good reason to change this?

Reply:
We'd need to keep Schema as a deprecated property, but I think we can rename it to path too.

("Schema" _only_ makes sense in the concept of HDFS/Hive which is where Airflow first supported. Even for a Mysql DB connection, schema doesn't make sense as a property name -- its a db name, not a schema)